app/controllers/steps_controller.rb

```diff
@@ -0,0 +1,56 @@
+class StepsController < ApplicationController
+  helper FormElementsHelper
+
+  def index
+    @prisoner_step = PrisonerStep.new
+    render :prisoner
+  end
+
+  def create
+    select_step
+  end
+
+private
+
+  def available_slots
+    @prisoner_step.prison.available_slots
+  end
+  helper_method :available_slots
```

Comment:
Reaching all the way into `@prisoner_step.prison.available_slots` from the view also feels wrong, but I'm not really happy about this. It feels like the right place would be `@slots_step.available_slots`, but that turns out to be quite a bit more intrusive to set up.